api/envoy/type/format.proto

```diff
@@ -0,0 +1,43 @@
+syntax = "proto3";
+
+package envoy.type;
+
+option java_outer_classname = "FormatProto";
+option java_multiple_files = true;
+option java_package = "io.envoyproxy.envoy.type";
+
+import "google/protobuf/struct.proto";
+
+// [#protodoc-title: Format]
+
+// Allows to define one of format: flat plain text or structured json.
+message StringOrJson {
+  oneof format {
+    // Allows to specify flat plain text format.
+    //
+    // .. code-block:: yaml
+    //
+    //   string_format: "My custom message"
+    //
+    string string_format = 1;
```

Comment:
should this field be `DataSource` to allow fetching body from the control plane/disk?

Reply:
Actually, it looks like `string_format` assumes formatting.
In this case my ask is to add 3rd option with non-formattable `DataSource`.

Reply:
I guess one reasonable question is "which API fields in general should be data sources?". Do we have a governing principle? Is it basically anything you might want to load from a disk? Do you have a concrete needs for this @euroelessar?

Reply:
> Do you have a concrete needs for this @euroelessar?

We want an ability to return custom hard-coded user-facing html page on errors generated by envoy.
In addition to that we want to be able to serve compile-time compressed version of it for clients which support gzip/brotli.

Combination of this two factors implies that we need an ability to return potentially large binary body.
Adding this bodies to the config itself is potentially non-practical, therefore the ask for `DataSource`.

To clarify, I'm fine with it not being part of this particular pull request, we can contribute the necessary pieces as follow ups, assuming the API can be extended to this needs.

Reply:
I think it would be nice to get the API clean here, what you're asking for isn't unreasonable. The actual code to consume from data source is a 1 liner, so if folks are in agreement let's do this.